modules/migrations/gitea_uploader.go

```diff
@@ -690,6 +690,21 @@ func (g *GiteaLocalUploader) newPullRequest(pr *base.PullRequest) (*models.PullR
 		}
 	} else {
 		head = pr.Head.Ref
+		// Ensure the closed PR SHA still points to an existing ref
+		_, err = git.NewCommand("rev-list", "-1", pr.Head.SHA).RunInDir(g.repo.RepoPath())
+		if err != nil {
+			if pr.Head.SHA != "" {
+				log.Warn("Deprecated local head, removing : %v", pr.Head.SHA)
+				relPath := "refs/pull/" + fmt.Sprintf("%d", pr.Number) + "/head"
```

Comment:
`pr.GetGitRefName()` ?

Reply:
Thanks for the suggestion ! 
I added the function in pullrequest.go 
`// GetGitRefName returns pull request relative path to head
func (p PullRequest) GetGitRefName() string {
	return fmt.Sprintf("refs/pull/%d/head", p.Number)
}`